pd/src/app.rs

```diff
@@ -74,6 +75,10 @@ pub struct App {
 
     /// Used to allow asynchronous requests to be processed sequentially.
     completion_tracker: CompletionTracker,
+
+    /// Contains the validator set, with each validator uniquely identified by their tendermint
+    /// public key.
+    validators: Arc<Mutex<BTreeMap<tendermint::PublicKey, Validator>>>,
```

Comment:
It looks like at the moment we don't need to share access to the `validators`, since it's set in the body of `init_genesis` where there's already a `&mut self` available. Are we expecting to need to read or write it in other places where we have to do asynchronous processing?

Reply:
I think we'll need to write to this in `end_block` (for validator updates) and `begin_block` (for handling evidence and rewards)

Reply:
Also non-blocking for this PR, but just a comment for the future: we might want to explore splitting out these fields into an `Inner` type, so that rather than having multiple chunks of synchronization, we can wrap all of the state that we need to access in one mutex.